tests/base/boring_model.py

```diff
@@ -122,10 +125,48 @@ def configure_optimizers(self):
         return [optimizer], [lr_scheduler]
 
     def train_dataloader(self):
-        return torch.utils.data.DataLoader(RandomDataset(32, 64))
+        return DataLoader(RandomDataset(32, 64))
 
     def val_dataloader(self):
-        return torch.utils.data.DataLoader(RandomDataset(32, 64))
+        return DataLoader(RandomDataset(32, 64))
 
     def test_dataloader(self):
-        return torch.utils.data.DataLoader(RandomDataset(32, 64))
+        return DataLoader(RandomDataset(32, 64))
+
+
+class BoringDataModule(LightningDataModule):
+    def __init__(self, data_dir: str = './'):
+        super().__init__()
+        self.data_dir = data_dir
+        self.non_picklable = None
+        self.checkpoint_state: Optional[str] = None
+
+    def prepare_data(self):
+        self.random_full = RandomDataset(32, 192)
+
+    def setup(self, stage: Optional[str] = None):
+        if stage == "fit" or stage is None:
+            self.random_train = Subset(self.random_full, indices=range(64))
+            self.random_val = Subset(self.random_full, indices=range(64, 128))
+            self.dims = self.random_train[0].shape
+
+        if stage == "test" or stage is None:
+            self.random_test = Subset(self.random_full, indices=range(128, 192))
+            self.dims = getattr(self, "dims", self.random_test[0].shape)
+
+        self.non_picklable = lambda x: x ** 2
```

Comment:
I think the tests that require `non_picklable` should subclass and override setup(), just as we do for BoringModel

Reply:
updated.